samples/star-wars-api/HttpHandlers.fs

```diff
@@ -37,22 +40,25 @@ module HttpHandlers =
 
     let private graphQL (next : HttpFunc) (ctx : HttpContext) =
         task {
+        
+            let logger = ctx.RequestServices.CreateLogger moduleType
 
             // TODO: validate the result
             let toResponse documentId =
                 function
                 | Direct   (data, errs) ->
+                    logger.LogInformation($"GraphQL toResponse Direct with documentId = '{documentId}', Data = {JsonSerializer.Serialize(data, Json.serializerOptions)}")
```

Comment:
You need to use structural logging here
```suggestion
                    logger.LogInformation("Produced direct GraphQL response with documentId = '{documentId}' and data = {data}", documented, data)
```
As I understand how it works, it will do log record creation only if log level is Information or above. So we must not do any heavy duty logic there. It will happen under the hood